### ISSUE TITLE
How to create a trainable variable in Keras

### ISSUE THREAD

**Issue by Derekkk** [keras-team/keras#10190]
Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps

- [x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

How can I create a trainable variables in Keras, like tf.get_variable() in tensorflow? I know in Keras, we can use K.Variable() to create a variable, but it seems that this function can only create non-trainable variable. So is there any way to do that？ Thanks for replying.

**Comment by paloha** [comment]
I am also interested in this. Have you found a solution please?

**Comment by bensteers** [comment]
Interestingly, I've been having the opposite problem (at least in `tensorflow==1.13.1`). It took me a bit to realize that the variable I was using to warm start a coefficient in my loss function was being trained. o.0

It seems that `K.variable` defaults to having `variable.trainable == True`, and if you assign your variable to a Layer instance attribute, it will be added to that layer's `trainable_weights/non_trainable_weights` property, based on the `variable.trainable` property.

That happens here:
https://github.com/tensorflow/tensorflow/blob/6612da89516247503f03ef76e974b51a434fb52e/tensorflow/python/keras/engine/base_layer.py#L1627

So here's what I'm doing as a work around to get non-trainable weights, but I think it's a bug that you can't just do `K.variable(..., trainable=False)`. I'm not sure what the effects of doing this are (I'm assuming there is some handling of `trainable` in `tf.Variable.__init__` and we're changing trainable after that has been called, so...), but when I do this it stops training it so I guess I'm going with it.

If your version on tf.keras has `variable.trainable == False` by default, you could perhaps try the opposite.

```python
class SampleLayer(Layer):
    def __init__(self, alpha, beta):
        # a trainable variable
        self.alpha = K.variable(alpha, name='alpha')

        # a non trainable variable
        beta = K.variable(beta, name='beta')
        beta._trainable = False
        self.beta = beta

        # or maybe you need to force it to be trainable
        beta = K.variable(beta, name='beta')
        beta._trainable = True
        self.beta = beta

    ...
```
